WalletWasabi.Tests/UnitTests/WabiSabi/Client/CoinJoinClientTests.cs

```diff
@@ -50,4 +54,43 @@ public void SanityCheckTest()
 			new[] { output2, output3 },
 			new[] { output1, AddOneSat(output2), SubOneSat(output3), output4 }));
 	}
+
+	[Fact]
+	public void GetTxOutsTest()
+	{
+		FeeRate feeRate = new(10m);
+
+		var outputs = new[]
+		{
+			Output.FromDenomination(Money.Coins(1m), ScriptType.P2WPKH, feeRate),
+			Output.FromDenomination(Money.Coins(2m), ScriptType.P2WPKH, feeRate),
+			Output.FromDenomination(Money.Coins(3m), ScriptType.Taproot, feeRate),
+			Output.FromDenomination(Money.Coins(4m), ScriptType.Taproot, feeRate),
+		};
+
+		var password = "satoshi";
+		var km = ServiceFactory.CreateKeyManager(password, true);
+		var destinationProvider = new InternalDestinationProvider(km);
+
+		var txOuts = CoinJoinClient.GetTxOuts(outputs, destinationProvider);
+
+		// All the outputs were generated.
+		Assert.Equal(txOuts.Count(), outputs.Length);
+
+		// No address reuse.
+		Assert.Distinct(txOuts.Select(x => x.ScriptPubKey));
+
+		// The count per script type is correct.
+		Assert.Equal(outputs.Count(x => x.ScriptType == ScriptType.Taproot), txOuts.Count(x => x.ScriptPubKey.IsScriptType(ScriptType.Taproot)));
+		Assert.Equal(outputs.Count(x => x.ScriptType == ScriptType.P2WPKH), txOuts.Count(x => x.ScriptPubKey.IsScriptType(ScriptType.P2WPKH)));
+
+		List<TxOut> toCheck = txOuts.ToList();
+		foreach (var output in outputs)
+		{
+			var foundTxOut = toCheck.First(txout => txout.ScriptPubKey.IsScriptType(output.ScriptType) && txout.Value == output.Amount);
```

Comment:
It would be okay to make it more strict and use `Single` instead of `First`.

Reply:
It is possible to have values with the same ScriptType and Amount.